### ISSUE TITLE
Bug: Perfmon requires Computer Name to remotely poll

### ISSUE THREAD

**Issue by TravisBoyle** [tableau/TabMon#99]
I've uninstalled tabmon and removed postgres, reinstalled using all defaults, as that was default issue resolution on previous access denied service start issues.

Now, I get the following:

```
2016-11-21 11:02:05,407 [5] INFO TabMon.Config.TabMonConfigReader - Loading TabMon user configuration..
2016-11-21 11:02:05,438 [5] DEBUG TabMon.Config.TabMonConfigReader - Loading database configuration..
2016-11-21 11:02:05,438 [5] INFO TabMon.Config.TabMonConfigReader - Connecting to results database..
2016-11-21 11:02:05,720 [5] DEBUG TabMon.Helpers.HostnameHelper - Successfully resolved 'edwtab2.cac.washington.edu' to 'edwtab2'.
2016-11-21 11:02:05,735 [5] INFO TabMon.Config.TabMonConfigReader - Successfully loaded TabMon config options! [Host='Primary\edwtab2', Writer='Database Writer (Postgres Driver)', PollInterval='60', TableName='countersamples']
2016-11-21 11:02:05,735 [5] INFO TabMon.TabMonAgent - Initializing TabMon..
2016-11-21 11:02:05,735 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading persistent performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-11-21 11:02:05,735 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-11-21 11:02:05,735 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Successfully validated 'Config\Counters.config' against 'Resources\CountersConfig.xsd'.
2016-11-21 11:02:05,735 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading Perfmon counters..
2016-11-21 11:02:10,345 [5] ERROR TabMon.TabMonAgent - Failed to initialize counter sampler using counters from configuration file: The network path was not found.

Aborting..
2016-11-21 11:02:10,345 [5] FATAL Topshelf.Runtime.Windows.WindowsServiceHost - The service did not start successfully
Topshelf.TopshelfException: The service did not start successfully (returned false).
   at Topshelf.Runtime.Windows.WindowsServiceHost.OnStart(String[] args)
2016-11-21 11:02:10,345 [1] ERROR Topshelf.HostFactory - The service terminated abnormally
System.NullReferenceException: Object reference not set to an instance of an object.
   at TabMon.TabMonAgent.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose()
   at Topshelf.Builders.ControlServiceBuilder`1.ControlServiceHandle.Dispose()
   at Topshelf.Runtime.Windows.WindowsServiceHost.Dispose(Boolean disposing)
   at System.ComponentModel.Component.Dispose()
   at System.ServiceProcess.ServiceBase.Run(ServiceBase[] services)
   at System.ServiceProcess.ServiceBase.Run(ServiceBase service)
   at Topshelf.Runtime.Windows.WindowsServiceHost.Run()
   at Topshelf.HostFactory.Run(Action`1 configureCallback)
```

Any assistance in getting tabmon installed and working is appreciated.

**Comment by jmangue** [comment]
Hi Travis,

It looks like TabMon is failing to resolve the hostname correctly.

Can you try changing the Host entry in TabMon.config to point to the IP address (instead of the name edwtab2) and let me know if it works?  If it does, it will confirm my theory and I can check in a fix for this.

Let me know how it goes!

**Comment by TravisBoyle** [comment]
updated to use IP address instead of hostname in tabmon.confg, same failure.

`2016-11-22 07:11:59,437 [5] INFO TabMon.Config.TabMonConfigReader - Loading TabMon user configuration..
2016-11-22 07:11:59,547 [5] DEBUG TabMon.Config.TabMonConfigReader - Loading database configuration..
2016-11-22 07:11:59,547 [5] INFO TabMon.Config.TabMonConfigReader - Connecting to results database..
2016-11-22 07:12:00,140 [5] DEBUG TabMon.Helpers.HostnameHelper - Successfully resolved '172.25.155.36' to 'edwtab2'.
2016-11-22 07:12:00,172 [5] INFO TabMon.Config.TabMonConfigReader - Successfully loaded TabMon config options! [Host='Primary\edwtab2', Writer='Database Writer (Postgres Driver)', PollInterval='60', TableName='countersamples']
2016-11-22 07:12:00,172 [5] INFO TabMon.TabMonAgent - Initializing TabMon..
2016-11-22 07:12:00,172 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading persistent performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-11-22 07:12:00,187 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-11-22 07:12:00,187 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Successfully validated 'Config\Counters.config' against 'Resources\CountersConfig.xsd'.
2016-11-22 07:12:00,187 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading Perfmon counters..
2016-11-22 07:12:04,812 [5] ERROR TabMon.TabMonAgent - Failed to initialize counter sampler using counters from configuration file: The network path was not found.

Aborting..
2016-11-22 07:12:04,812 [5] FATAL Topshelf.Runtime.Windows.WindowsServiceHost - The service did not start successfully
Topshelf.TopshelfException: The service did not start successfully (returned false).
   at Topshelf.Runtime.Windows.WindowsServiceHost.OnStart(String[] args)
2016-11-22 07:12:04,859 [1] ERROR Topshelf.HostFactory - The service terminated abnormally
System.NullReferenceException: Object reference not set to an instance of an object.
   at TabMon.TabMonAgent.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose()
   at Topshelf.Builders.ControlServiceBuilder`1.ControlServiceHandle.Dispose()
   at Topshelf.Runtime.Windows.WindowsServiceHost.Dispose(Boolean disposing)
   at System.ComponentModel.Component.Dispose()
   at System.ServiceProcess.ServiceBase.Run(ServiceBase[] services)
   at System.ServiceProcess.ServiceBase.Run(ServiceBase service)
   at Topshelf.Runtime.Windows.WindowsServiceHost.Run()
   at Topshelf.HostFactory.Run(Action`1 configureCallback)`

**Comment by jmangue** [comment]
It looks like we have a bug here.. both the IP and your longer DNS entry are being resolved and then overly shortened down to simply "edwtab2", which Perfmon can't connect to; we need to preserve the domain suffix information for cases like this.

We can probably turn around a fix for this fairly quickly.. I'll keep you posted here.

**Comment by sycogne** [comment]
Hi Travis, Hi Jmangue

I currently experience the same issue on a distributed environment where TabMon is installed on another server as Tableau Server. Did you find a solution to this problem?

2016-12-13 10:32:28,683 [5] INFO TabMon.Config.TabMonConfigReader - Loading TabMon user configuration..
2016-12-13 10:32:28,715 [5] DEBUG TabMon.Config.TabMonConfigReader - Loading database configuration..
2016-12-13 10:32:28,715 [5] INFO TabMon.Config.TabMonConfigReader - Connecting to results database..
2016-12-13 10:32:28,871 [5] DEBUG TabMon.Helpers.HostnameHelper - Successfully resolved 'sedcatab0010.emea.ebs.corpintra.net' to 'sedcatab0010'.
2016-12-13 10:32:28,871 [5] INFO TabMon.Config.TabMonConfigReader - Successfully loaded TabMon config options! [Host='Primary\sedcatab0010', Writer='Database Writer (Postgres Driver)', PollInterval='60', TableName='countersamples']
2016-12-13 10:32:28,871 [5] INFO TabMon.TabMonAgent - Initializing TabMon..
2016-12-13 10:32:28,886 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading persistent performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-12-13 10:32:28,886 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading performance counters from C:\Program Files (x86)\TabMon\Config\Counters.config..
2016-12-13 10:32:28,886 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Successfully validated 'Config\Counters.config' against 'Resources\CountersConfig.xsd'.
2016-12-13 10:32:28,886 [5] DEBUG TabMon.CounterConfig.CounterConfigLoader - Loading Perfmon counters..
2016-12-13 10:33:10,919 [5] ERROR TabMon.TabMonAgent - Failed to initialize counter sampler using counters from configuration file: The network path was not found.

Aborting..
2016-12-13 10:33:10,919 [5] FATAL Topshelf.Runtime.Windows.WindowsServiceHost - The service did not start successfully
Topshelf.TopshelfException: The service did not start successfully (returned false).
   at Topshelf.Runtime.Windows.WindowsServiceHost.OnStart(String[] args)
2016-12-13 10:33:10,919 [1] ERROR Topshelf.HostFactory - The service terminated abnormally
System.NullReferenceException: Object reference not set to an instance of an object.
   at TabMon.TabMonAgent.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose(Boolean disposing)
   at TabMonService.TabMonServiceBootstrapper.Dispose()
   at Topshelf.Builders.ControlServiceBuilder`1.ControlServiceHandle.Dispose()
   at Topshelf.Runtime.Windows.WindowsServiceHost.Dispose(Boolean disposing)
   at System.ComponentModel.Component.Dispose()
   at System.ServiceProcess.ServiceBase.Run(ServiceBase[] services)
   at System.ServiceProcess.ServiceBase.Run(ServiceBase service)
   at Topshelf.Runtime.Windows.WindowsServiceHost.Run()
   at Topshelf.HostFactory.Run(Action`1 configureCallback)

**Comment by TravisBoyle** [comment]
I've had to back burner the TabMon implementation for now, with the intent that I would get back to it after the bug fix jmangue noted above was available.

**Comment by CapoD** [comment]
Hi Jmangue,

I was curious wether the bug fixing made any progress?

best wishes

**Comment by jmangue** [comment]
Hey all, sorry for the slow response on this one.  I've been researching this... it seems that the C# library for working with Perfmon requires that the target machine be on the same domain for remote polling.

The fix is not as easy as I initially thought, but I'll keep digging at it.

In the meantime, a workaround for this would be to ensure that TabMon is installed on a host on the same domain, and then specify the hostname in the TabMon config to be the simple computer name of the target machine.

**Comment by seanloftus** [comment]
Hey Jmangue - 

I'm having the same issues it seems.  I've been trying to install TABMON and I'm getting the 
"Error 5: Access is Denied" when starting the TabMon Service.  Has there been any updates on this issue?

**Comment by seanloftus** [comment]
I've also confirmed that all of my machines are installed on the same domain.  So that's not the issue here either.

**Comment by danjrahm** [comment]
This issue has been released by v1.2. Closing now.
